ports/apr/unofficial-apr-config.cmake

```
@@ -0,0 +1,17 @@
+message(WARNING "find_package(unofficial-apr) is deprecated.\nUse find_package(apr) instead")
+include(CMakeFindDependencyMacro)
+find_dependency(apr CONFIG)
+
+if("${VCPKG_LIBRARY_LINKAGE}" STREQUAL "static")
+    add_library(unofficial::apr::apr-1 INTERFACE IMPORTED)
+    target_link_libraries(unofficial::apr::apr-1 INTERFACE apr::apr-1)
+
+    add_library(unofficial::apr::aprapp-1 INTERFACE IMPORTED)
+    target_link_libraries(unofficial::apr::aprapp-1 INTERFACE apr::aprapp-1)
+else()
+    add_library(unofficial::apr::libapr-1 INTERFACE IMPORTED)
+    target_link_libraries(unofficial::apr::libapr-1 INTERFACE apr::libapr-1)
+
+    add_library(unofficial::apr::libaprapp-1 INTERFACE IMPORTED)
+    target_link_libraries(unofficial::apr::libaprapp-1 INTERFACE apr::libaprapp-1)
+endif()

```

Build-time triplet variable `VCPKG_LIBRARY_LINKAGE` is unknown at config use time, in project mode.
In addition, it is necessary to check if the target already exists.

```suggestion
if(TARGET apr::apr-1 AND NOT TARGET unofficial::apr::apr-1)
    add_library(unofficial::apr::apr-1 INTERFACE IMPORTED)
    target_link_libraries(unofficial::apr::apr-1 INTERFACE apr::apr-1)
elseif(TARGET apr::libapr-1 AND NOT TARGET unofficial::apr::libapr-1)
    add_library(unofficial::apr::libapr-1 INTERFACE IMPORTED)
    target_link_libraries(unofficial::apr::libapr-1 INTERFACE apr::libapr-1)
endif()
if(TARGET apr::aprapp-1 AND NOT TARGET unofficial::apr::aprapp-1)
    add_library(unofficial::apr::aprapp-1 INTERFACE IMPORTED)
    target_link_libraries(unofficial::apr::aprapp-1 INTERFACE apr::aprapp-1)
elseif(TARGET apr::libaprapp-1 AND NOT TARGET unofficial::apr::libaprapp-1)
    add_library(unofficial::apr::libaprapp-1 INTERFACE IMPORTED)
    target_link_libraries(unofficial::apr::libaprapp-1 INTERFACE apr::libaprapp-1)
endif()
```

I wish the upstream target name wouldn't change with library linkage, but I guess this ship has sailed.

---

Ops, didn't notice. Pushed the fix as https://github.com/microsoft/vcpkg/pull/40742/commits/ce0ef6f49cf3a6741a97ec63257373f5db890ee0.

I think there should be the check for the target existing, because otherwise the target will be created, but will work incorrectly.